fix(expo-go): prevent non-app expo.dev subdomains from opening in Expo Go

hassankhan opened the pull request:
# Why

Excludes subdomains from Expo Go's internal URL handling. These should open in the external browser instead of being treated as Expo projects.

The issue does not occur on Android.

<!--
Please describe the motivation for this PR, and link to relevant GitHub issues, forums posts, or feature requests.
-->

# How

Added a new check in `isExpoHostedUrlComponents` to match the URL host against our known non-app subdomains.

<!--
How did you build this feature or fix this bug and why?
-->

# Test Plan

| Before | After |
|--|--|
| ![RocketSim_Recording_iPhone_16_Pro_6 3_2025-08-21_22 43 38](https://github.com/user-attachments/assets/60a7f8ab-c2ed-4bed-a0a1-73e06766c8ee) | ![RocketSim_Recording_iPhone_16_Pro_6 3_2025-08-21_22 31 50](https://github.com/user-attachments/assets/fa56c015-94e2-425f-ba75-190afa1d0852) | 


<!--
Please describe how you tested this change and how a reviewer could reproduce your test, especially if this PR does not include automated tests! If possible, please also provide terminal output and/or screenshots demonstrating your test/reproduction.
-->

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [x] I added a `changelog.md` entry and rebuilt the package sources according to [this short guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

hassankhan commented on apps/expo-go/ios/Exponent/Versioned/Core/Api/EXUtil.m:
```diff
@@ -38,6 +38,17 @@ + (NSString *)escapedResourceName:(NSString *)name
   return [name stringByAddingPercentEncodingWithAllowedCharacters:allowedCharacters];
 }
 
++ (BOOL)isExcludedExpoHost:(NSString *)host
```
If preferred, we could invert this logic and instead only _include_ `u.expo.dev`; I wasn't sure if there were any other subdomains.

brentvatne commented on apps/expo-go/ios/Podfile.lock:
```diff
@@ -4284,7 +4284,7 @@ SPEC CHECKSUMS:
   GoogleMaps: 8939898920281c649150e0af74aa291c60f2e77d
   GoogleUtilities: 26a3abef001b6533cf678d3eb38fd3f614b7872d
-  hermes-engine: e4b02512e9e8b7e4f91059d416c8e1269f78170d
+  hermes-engine: bda77156b2a932908c70191b9e35f27462596760
```
you can revert this change

hassankhan committed:
fix(expo-go): prevent non-app expo.dev subdomains from opening in Expo Go

Excludes subdomains from Expo Go's internal URL handling. These should open in the external browser instead of being treated as Expo projects.
